frontend/packages/metal3-plugin/src/feature.ts

```diff
@@ -0,0 +1,44 @@
+import { k8sGet, K8sResourceKind } from '@console/internal/module/k8s';
+import { InfrastructureModel } from '@console/internal/models';
+import { getInfrastructurePlatform } from '@console/shared/src/selectors/infrastructure';
+import * as _ from 'lodash';
+import { BAREMETAL } from './constants';
+
+const retryFlagDetection = (dispatch, cb) => {
+  setTimeout(() => cb(dispatch), 15000);
+};
+
+const handleError = (res, flag, dispatch, cb) => {
+  const status = _.get(res, 'response.status');
+  if (_.includes([403, 502], status)) {
+    dispatch({ type: 'addFlag', payload: { flag } });
+  }
+  if (!_.includes([401, 403, 500], status)) {
+    retryFlagDetection(dispatch, cb);
+  }
+};
+
+export const detectBaremetalPlatform = (dispatch) => {
+  dispatch({ type: 'addFlag', payload: { flag: BAREMETAL } });
```

Comment:
We should look into why we can't reuse the existing `detectBaremetalPlatform` function as-is.

`k8sGet`, `setFlag` and `handleError` should be import-able from `@console/internal` package.